src/posix/main.c

```diff
@@ -133,48 +132,62 @@ static const struct option kOptions[] = {{"debug-level", required_argument, NULL
-            "        --spi-small-packet=[n]    Specify the smallest packet we can receive in a single transaction.\n"
-            "                                  (larger packets will require two transactions). Default value is 32.\n"
+static const char *sHelpString =
+    "Syntax:\n"
+    "    %s [Options] NodeId|Device|Command [DeviceConfig|CommandArgs]\n"
+    "Options:\n"
+    "    -I  --interface-name name     Thread network interface name.\n"
+    "    -d  --debug-level             Debug level of logging.\n"
+    "    -n  --dry-run                 Just verify if arguments is valid and radio spinel is compatible.\n"
+    "        --no-reset                Do not send Spinel reset command to RCP on initialization.\n"
+    "        --radio-version           Print radio firmware version.\n"
+    "        --ncp-dataset             Retrieve and save NCP dataset to file.\n"
+    "    -s  --time-speed factor       Time speed up factor.\n"
+    "    -v  --verbose                 Also log to stderr.\n"
+    "        --gpio-int-dev[=gpio-device-path]\n"
+    "                                  Specify a path to the Linux sysfs-exported GPIO device for the\n"
+    "                                  `I̅N̅T̅` pin. If not specified, `SPI` interface will fall back to\n"
+    "                                  polling, which is inefficient.\n"
+    "        --gpio-int-line[=line-offset]\n"
+    "                                  The offset index of `I̅N̅T̅` pin for the associated GPIO device.\n"
+    "                                  If not specified, `SPI` interface will fall back to polling,\n"
+    "                                  which is inefficient.\n"
+    "        --gpio-reset-dev[=gpio-device-path]\n"
+    "                                  Specify a path to the Linux sysfs-exported GPIO device for the\n"
+    "                                  `R̅E̅S̅` pin.\n"
+    "        --gpio-reset-line[=line-offset]"
+    "                                  The offset index of `R̅E̅S̅` pin for the associated GPIO device.\n"
+    "        --spi-mode[=mode]         Specify the SPI mode to use (0-3).\n"
+    "        --spi-speed[=hertz]       Specify the SPI speed in hertz.\n"
+    "        --spi-cs-delay[=usec]     Specify the delay after C̅S̅ assertion, in µsec.\n"
+    "        --spi-reset-delay[=ms]    Specify the delay after R̅E̅S̅E̅T̅ assertion, in milliseconds.\n"
+    "        --spi-align-allowance[=n] Specify the maximum number of 0xFF bytes to clip from start of\n"
+    "                                  MISO frame. Max value is 16.\n"
+    "        --spi-small-packet=[n]    Specify the smallest packet we can receive in a single transaction.\n"
+    "                                  (larger packets will require two transactions). Default value is 32.\n"
+    "    -h  --help                    Display this usage information.\n";
+#else
+static const char *sHelpString =
+    "Syntax:\n"
```

Comment:
The code between line 139 and 149 and the code between line 174 and 184 are the same. Could we define a common help string to represent it? Otherwise, we need to update two places when we want to modify the common help string in the future.